ouroboros-network/src/Ouroboros/Network/TxSubmission/Inbound.hs

```diff
@@ -337,47 +339,63 @@ txSubmissionInbound _tracer maxUnacked mpReader mpWriter _version =
     -- need to bother requesting from the client since they're already in the
     -- mempool.
     --
-    acknowledgeTxIdsInMempool :: ServerState txid tx
+    acknowledgeTxIds :: ServerState txid tx
+                              -> StrictSeq txid
+                              -> Map txid TxSizeInBytes
                               -> MempoolSnapshot txid tx idx
                               -> ServerState txid tx
-    acknowledgeTxIdsInMempool st MempoolSnapshot{mempoolHasTx} =
+    acknowledgeTxIds st txidsSeq _ _ | Seq.null txidsSeq  = st
+    acknowledgeTxIds st txidsSeq txidsMap MempoolSnapshot{mempoolHasTx} =
         -- Return the next 'ServerState'
         st {
           availableTxids      = availableTxids',
           bufferedTxs         = bufferedTxs'',
-          unacknowledgedTxIds = unacknowledgedTxIds',
+          unacknowledgedTxIds = unacknowledgedTxIds'',
           numTxsToAcknowledge = numTxsToAcknowledge st
                               + fromIntegral (Seq.length acknowledgedTxIds)
         }
       where
-        -- Divide the available txs in two: those that are already in the
-        -- mempool and those that are not. We'll request some txs from the
+
+        (ignoredTxids, availableTxidsMp) =
+              Map.partitionWithKey
+                (\txid _ -> mempoolHasTx txid)
+                txidsMap
+
+        availableTxidsU =
+              Map.filterWithKey
+                (\txid _ -> not $ elem txid (unacknowledgedTxIds st))
+                txidsMap
+
+        -- Divide the new txids in two: those that are already in the
+        -- mempool or in flight and those that are not. We'll request some txs from the
```

Comment:
This comment seems out of place now, I think it ought to be at the top of this hunk.